twine/__main__.py

```diff
@@ -24,9 +25,19 @@
 
 def main() -> Any:
     try:
-        return cli.dispatch(sys.argv[1:])
-    except (exceptions.TwineException, requests.HTTPError) as exc:
-        return _format_error(f"{exc.__class__.__name__}: {exc.args[0]}")
+        result = cli.dispatch(sys.argv[1:])
+    except requests.HTTPError as exc:
+        status_code = exc.response.status_code
+        status_phrase = http.HTTPStatus(status_code).phrase
+        result = (
+            f"{exc.__class__.__name__} from {exc.response.url}: "
+            f"{status_code} {status_phrase}\n"
+            f"{exc.response.reason}"
```

Comment:
I don't think the URL is important enough to put at the start of the first line. Having the status code right after HTTPError feels like better form to me -- maybe it's just me, but I was a bit surprised that the new format doesn't present the error code first up.

I am 100% on board for moving the server-response to a newline. With that in mind, how about...

```suggestion
            f"{exc.__class__.__name__}: {status_code} {status_phrase} (from {exc.response.url})\n"
            f"{exc.response.reason}"
```

This gives us:

```
HTTPError: 403 Client Error (from https://test.pypi.org/legacy/)
Invalid or non-existent authentication information. See https://test.pypi.org/help/#invalid auth for details
```

Reply:
Thanks for the feedback, @pradyunsg. Personally, I think the source of the error (i.e. the repository, e.g. PyPI) is quite useful when troubleshooting. Making it more prominent was my main motivation for doing this work. With the descriptive server response on the second, I think the status code and phrase is less important.

Reply:
Would it be OK to put the URL on the second line, making the error message a 3 line thing? (move from {url} to the next line and rephrase).

As someone familiar with both twine and HTTP, I certainly feel that the url (which can be arbitrarily long, if you have a custom index) shouldn't come before the status codes on the same line.

(to be clear, I'm only providing feedback as an "expert user")

Reply:
After playing around with this a bit, I took your first suggestion, but without the parens. I put a new screenshot in the description.

```
HTTPError: 403 Forbidden from https://test.pypi.org/legacy/
Invalid or non-existent authentication information. See https://test.pypi.org/help/#invalid-auth for more information.
```